src/precice/impl/Participant.hpp

```diff
@@ -73,98 +72,178 @@ class Participant {
       const mesh::PtrData &data,
       const mesh::PtrMesh &mesh);
 
+  void addReadMappingContext(MappingContext *mappingContext);
+
+  void addWriteMappingContext(MappingContext *mappingContext);
+
+  void addWatchPoint(const PtrWatchPoint &watchPoint);
+
+  void addWatchIntegral(const PtrWatchIntegral &watchIntegral);
+
+  void setUseMaster(bool useMaster);
+
+  void setMeshIdManager(std::unique_ptr<utils::ManageUniqueIDs> &&idm)
+  {
+    _meshIdManager = std::move(idm);
+  }
+
+  void addAction(action::PtrAction &&action);
+
+  /// Adds an export context to export meshes and data.
+  void addExportContext(const io::ExportContext &context);
+
+  /// Adds a mesh to be used by the participant.
+  void useMesh(
+      const mesh::PtrMesh &                         mesh,
+      const Eigen::VectorXd &                       localOffset,
+      bool                                          remote,
+      const std::string &                           fromParticipant,
+      double                                        safetyFactor,
+      bool                                          provideMesh,
+      partition::ReceivedPartition::GeometricFilter geoFilter);
+  /// @}
+
+  /// @name Data querries
+  /// @{
+  /// Provides access to both write and read \ref DataContext
   const DataContext &dataContext(int dataID) const;
 
+  /// Provides access to both write and read \ref DataContext
   DataContext &dataContext(int dataID);
 
+  /// Provides access to write \ref DataContext
   const utils::ptr_vector<DataContext> &writeDataContexts() const;
 
+  /// Provides access to write \ref DataContext
   utils::ptr_vector<DataContext> &writeDataContexts();
 
+  /// Provides access to read \ref DataContext
   const utils::ptr_vector<DataContext> &readDataContexts() const;
 
+  /// Provides access to read \ref DataContext
   utils::ptr_vector<DataContext> &readDataContexts();
 
-  bool isMeshUsed(int meshID) const;
-
-  bool isMeshProvided(int meshID) const;
+  /// Is the dataID know to preCICE?
+  bool hasData(int dataID) const;
 
+  /// Is the data used by this participant?
   bool isDataUsed(int dataID) const;
 
+  /// Is the data used by this participant?
+  bool isDataUsed(const std::string &dataName, int meshId) const;
+
+  /// Is the participant allowed to read the data?
   bool isDataRead(int dataID) const;
 
+  /// Is the participant allowed to write the data?
   bool isDataWrite(int dataID) const;
 
+  /// What is the dataID of the used data from a used mesh given the meshid and the data name?
+  int getUsedDataID(const std::string &dataName, int meshID) const;
+
+  /// What is the name of the given data id
+  std::string getDataName(int dataID) const;
+  /// @}
+
+  /// @name Mesh querries
+  /// @{
+  /// Provides access to all \ref MeshContext where the index is meshID. Can contain nullptr!
   const MeshContext &meshContext(int meshID) const;
```

Comment:
Could you add some more documentation here? What does it mean when I get `nullptr` and how to interpret it. Is it a misuse?!

Reply:
Thanks for the hint! The documentation was incorrect.